qwandaq/src/main/java/life/genny/qwandaq/AnswerLink.java

```diff
@@ -198,168 +200,141 @@ public AnswerLink(final BaseEntity source, final BaseEntity target, final Answer
 	 */
 	@JsonIgnore
 	public void setAnswer(final Answer answer) {
+
 		this.setCreated(answer.getCreated());
 		this.setExpired(answer.getExpired());
 		this.setRefused(answer.getRefused());
 		this.setInferred(answer.getInferred());
 		
-		List<String> formatStrings = null;
-
-		String className = "";
-		try {
-			className = this.getAttribute().getDataType().getClassName();
-		} catch (NullPointerException e) {
-			e.printStackTrace();
+		Attribute attr = this.getAttribute();
+		DataType dataType = attr.getDataType();
+		String className = dataType.getClassName();
+		String result = answer.getValue();
+
+		if (StringUtils.isBlank(result) || ("null".equals(result) && !className.contains("String"))) {
+			valuesNull();
+			return;
 		}
 
 		switch (className) {
-			case "life.genny.qwandaq.entity":
-				List<String> beCodeList = new CopyOnWriteArrayList<String>();
-				beCodeList.add(answer.getValue());
-				setValueBaseEntityCodeList(beCodeList);
+
+			case "life.genny.qwandaq.entity.BaseEntity":
+			case "life.genny.qwanda.entity.BaseEntity":
```

Comment:
do we need both? only qwandaq variant would suffice?

Reply:
Believe it or not, there are cases in internmatch where the old qwanda version surfaces